glide-core/src/client/value_conversion.rs

```diff
@@ -182,6 +183,38 @@ pub(crate) fn convert_to_expected_type(
             )
                 .into()),
         },
+        // command returns nil or an array of 2 elements, where the second element is a map represented by a 2D array
+        // we convert that second element to a map as we do in `MapOfStringToDouble`
+        /*
+        127.0.0.1:6379> zmpop 1 z1 min count 10
+        1) "z1"
+        2) 1) 1) "2"
+              2) (double) 2
+           2) 1) "3"
+              2) (double) 3
+         */
+        ExpectedReturnType::ZMPopReturnType => match value {
+            Value::Nil => Ok(value),
+            Value::Array(array) if array.len() == 2 && matches!(array[1], Value::Array(_)) => {
```

Comment:
BZMPOP can return more than 1 key and more than 2 items in the array, right? 
Or does it always return only one key (even if multiple keys are supplied and COUNT)

Reply:
only 1 key, but could be multiple elements